### ISSUE TITLE
CFE: support for loops in collections breaks set/map disambiguation

### ISSUE THREAD

**Issue by kmillikin** [dart-lang/sdk#36332]
The change at [CL 97628](https://dart-review.googlesource.com/c/sdk/+/97628) will break a pair of co19 tests:
```
co19_2/LanguageFeatures/Set-literals/disambiguating_A06_t01/06 broke (CompileTimeError -> Pass, expected CompileTimeError)
co19_2/LanguageFeatures/Set-literals/disambiguating_A06_t01/07 broke (CompileTimeError -> Pass, expected CompileTimeError)
```
These failures happen under the experimental flag `control-flow-collections`.  They should be investigated and fixed before the feature is launched.

**Comment by kmillikin** [comment]
This is expected, because we need code to explicitly handle the control flow cases.

**Comment by stefantsov** [comment]
It is fixed in https://dart-review.googlesource.com/c/sdk/+/98015.  I think we may close this issue when CL 98015 lands.

**Comment by stefantsov** [comment]
Fixed in 584affcffa3d893a99938cceebc312a76fb82934 and follow-ups.
